### ISSUE TITLE
Unable connect to datasource via direct access

### ISSUE THREAD

**Issue by jsedy7** [grafana/grafana#10863]
Tested versions: **Grafana 5.0.0-11668beta1** and **Grafana 4.6.3**
Datasource: Prometheus
OS: Docker container (Debian)

I have a problem with a connection to my Prometheus via Direct access. (see Screenshot)

![grafana5_noproxy](https://user-images.githubusercontent.com/8748774/36029680-82992aaa-0da4-11e8-9955-f97010b4177b.png)

In the last version of Grafana that wasn't a problem. It worked correctly. Is anything different in newest version of Grafana? 
![grafana4 3 6](https://user-images.githubusercontent.com/8748774/36029688-870daba6-0da4-11e8-87c9-4d4e8445b215.png)

If I use proxy access, it works correctly for this datasource, but different datasource in my different network not works. 
![grafana5_proxy](https://user-images.githubusercontent.com/8748774/36029687-86ebfcd6-0da4-11e8-98d9-07b996e9085d.png)

I set up a debug logging, but I don't see nothing wrong in logs. Could you help me with this situation? 
_I use environment variables as http_proxy and https_proxy which I use for notification channels as Slack._  

Thank you! 🙂

**Comment by torkelo** [comment]
please use community site for questions, https://community.grafana.com, 

You probably have a networking issue, not sure how we can help

**Comment by jsedy7** [comment]
I tested it again with direct access (see screenshot) and I tested using datasource on specific graph. I wanted to know, If graph shows something. Datasource works correctly, but after saving datasource configuration still shows exclamation mark. I do not know why, but I think, that's not correct. 🙂 It's confusing.. 🙂 

![grafana5_noproxy](https://user-images.githubusercontent.com/8748774/36030784-ccb334d8-0da8-11e8-8fee-4fe230031a3d.png)

**Comment by torkelo** [comment]
open chrome dev tools console, any errors there?

**Comment by jsedy7** [comment]
So, I tested it again with direct access and I tested using datasource on specific graph. I wanted to know, If graph shows something. Datasource don't work correctly, after saving datasource configuration still shows exclamation mark and graph is still loading on this datasource. 

After saving datasource is in browser console (Firefox): 
`Blocked loading mixed active content “http://myserver:9090/api/v1/label/__name__/values”
[Learn More]`

Browser console (Chrome):
`angular.js:12837 Mixed Content: The page at 'https://mygrafana:3002/datasources/edit/2' was loaded over HTTPS, but requested an insecure XMLHttpRequest endpoint 'http://myprometheus:9090/api/v1/label/__name__/values'. This request has been blocked; the content must be served over HTTPS.`

I can do curl to my prometheus and it return metrics correctly. With using proxy access it works, with direct access not works... So I switched from https to http protocol, commented configuration with certificate paths and after this change everything works correctly. 

![screenshot_20180209_161505](https://user-images.githubusercontent.com/8748774/36034484-81044e76-0db4-11e8-9ea5-f958c2c5e867.png)

Thank you I appreciate your help. @torkelo
